packages/uswds-core/src/styles/mixins/typography/typeset.scss

```diff
@@ -55,12 +51,8 @@ Sets:
   margin-top: 0;
   max-width: measure($theme-text-measure);
 
-  * + & {
-    margin-top: 1em; // TODO: add as var to settings?
-  }
```

Comment:
1. Yes, this was the beginning of the problems. It would seem to bubble the selector all the way up and incorrectly attach styles to classes. We moved the `* + & {}` into [usa-prose](https://github.com/uswds/uswds/blob/cbf51140ebe8b6af264e60668d6deabba03cb008/packages/usa-prose/src/styles/_usa-prose.scss#L12-L24) to resolve this.
2. I am unaware of other selectors adding additional settings as a result of this compilation error. If you have an example we can dive deeper!

Reply:
1. Are there examples of this?
2. I mean, is this why this selector was moved to component classes like hero and graphic-list

Reply:
@mejiaj I am not sure if this helps, but there are some details about the  `* + &` issue `typeset` from all the way back in [this comment](https://github.com/uswds/uswds/pull/4576#issuecomment-1065577521) (This issue has been a journey), specifically in relation to the CSS output for items like ` * + .usa-prose > h1`.

After discussion, the best solution seemed to be to just pull the `* + &` styles from all mixins and instead add those rules explicitly wherever the mixins are called.

Reply:
@mahoneycm @amyleadem thanks for linking that — that does help! The main issue is `* + &` being nested in a selector (like in usa-prose):

```scss
.usa-prose {
  @include typeset($theme-prose-font-family);

  // The `& > ` causes unexpected behavior.
  & > {
    @include usa-paragraph-style;
    @include usa-headings-styles;
    @include usa-list-styles;
    @include usa-table-styles;
  }
}
```

I also had several attempts at using SASS' [`@at-root`](https://sass-lang.com/documentation/at-rules/at-root) without luck. What do you think of a compromise that reverts `typeset-p` and `typeset-heading` and it's use in components, but does something different for `usa-prose`?

Something like:

```scss
.usa-prose {
  @include typeset($theme-prose-font-family);

  & > {
    @include usa-list-styles;
    @include usa-table-styles;
  }

  & > * + p,
  & > p + * {
      // Margin
  }
}

…etc
```

This way we minimize the amount of styles to be added manually to each component.

Reply:
I'll try this out today and compare the CSS output. I'll also check it with the test repo provided for #4962 to see if the build error still happens.